python/pyspark/sql/tests/test_session.py

```diff
@@ -103,18 +103,18 @@ def test_SparkSession(self):
             .master("local") \
             .config("some-config", "v2") \
             .getOrCreate()
-        try:
-            self.assertEqual(spark.conf.get("some-config"), "v2")
-            self.assertEqual(spark.sparkContext._conf.get("some-config"), "v2")
-            self.assertEqual(spark.version, spark.sparkContext.version)
-            spark.sql("CREATE DATABASE test_db")
-            spark.catalog.setCurrentDatabase("test_db")
-            self.assertEqual(spark.catalog.currentDatabase(), "test_db")
-            spark.sql("CREATE TABLE table1 (name STRING, age INT) USING parquet")
-            self.assertEqual(spark.table("table1").columns, ['name', 'age'])
-            self.assertEqual(spark.range(3).count(), 3)
-        finally:
-            spark.stop()
+        with self.database("test_db"):
```

Comment:
Is it because the database wasn't cleaned? In that case, you can just add `class SparkSessionTests3(unittest.TestCase, SQLTestUtils):`.

Reply:
SQLTestUtils excepts spark session to be present [Code here](https://github.com/apache/spark/blob/master/python/pyspark/testing/sqlutils.py#L196). So have ReusedSQLTestCase which will create spark session and inherits `unittest.TestCase, SQLTestUtils`

Reply:
This set of tests are specific to Spark sessions. Let's don't make another Spark session before the tests. You can just manually drop the database explicitly in finally

Reply:
Handled